### PR TITLE
Refactor operations in All_in_One_SEO_Pack_Opengraph::add_meta()

### PR DESCRIPTION
**Pull request by EkoJR**
Issue #2938

## Proposed changes

Refactors operations in `All_in_One_SEO_Pack_Opengraph::add_meta()` to fix a number problems that made tracking variables and debugging difficult.

1. Refactors variable variables to be direct variables.
2. Renaming foreach variables from single characters to a proper name that better defines its use.
3. Change to group operations related to each other. Currently some operations are spread out within the function/method.
4. Unnecessarily using a filter to access a method within the same class, and the filter has only a single use. Changing to use the method directly makes it easier to track/inspect.
5. Add missing Filter Documentation.

## Types of changes

What types of changes does your code introduce?

- Improves existing functionality
- Improves existing code

## Checklist

- [x] I've selected the appropriate branch (ie x.y rather than master).
- [x] I've created an appropriate title, descriptive of what the PR does.
- [x] Travis-ci runs with no errors.
- [ ] I have added tests that prove my fix is effective/my feature works or have created an issue for it (if appropriate).
- [ ] I have added necessary documentation, or have created an issue for docs (if appropriate).

## Testing instructions
Code refactor, so no changes to expected output. Testing would primarily be on the frontend.

1. Enable the Opengraph Module.
2. Ensure CTPs are enabled on content you are testing (ex. Post and Pages).
3. Any options that may effect output can be changed as desired.
4. (Optional) Go to edit post, and modify any Opengraph settings.
5. Go to frontend and check for any errors or missing data.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wpsmort** [comment]
@EkoJR I tested this and found one problem.  If the site is set to HTTPS, the og:image:secure_url and og:video:secure_url meta tags are not correctly set and the following notices get output when viewing the site:

Undefined index: thumbnail_1 in aioseop_opengraph.php on line 1667
Undefined index: thumbnail_1 in aioseop_opengraph.php on line 1568
Undefined index: video_1 in aioseop_opengraph.php on line 1667
Undefined index: video_1 in aioseop_opengraph.php on line 1568

The meta tags looks like this (i.e. they are missing the URL for the image/video and the property is incorrect):

<meta property="thumbnail_1" content="og:image:secure_url" />
<meta property="video_1" content="og:video:secure_url" />

To reproduce this, follow these steps:

1. Create a new site that has HTTPS enabled
2. Install the PR code and activate the Social Meta module
3. Visit the front end and you should see the notices

**Comment by EkoJR** [comment]
Nice catch @wpsmort 

I added changes for the an SSL enabled site. Should be good now.

**Comment by wpsmort** [comment]
This is tested and is good.

File: modules/aioseop_opengraph.php [hunk 1]
```diff
@@ -1502,60 +1465,55 @@ function add_meta() {
...
-					'title'             => 'twitter:title',
-					'description'       => 'twitter:description',
-					'twitter_thumbnail' => 'twitter:image',
+					'twitter:card'        => $card,
+					'twitter:site'        => $site,
+					'twitter:creator'     => $creator,
+					'twitter:domain'      => $domain,
+					'twitter:title'       => $title,
+					'twitter:description' => $description,
+					'twitter:image'       => $twitter_thumbnail,
```
Comment (arnaudbroes) [inline modules/aioseop_opengraph.php]: This code occurs three times. Wouldn't it be better to pull this into one or two separate functions?

**Comment by arnaudbroes** [comment]
@EkoJR just one small comment.

File: modules/aioseop_opengraph.php (hunk 1 above)
Comment (EkoJR) [reply modules/aioseop_opengraph.php]: I believe you are referring to the reference array that occurs two times. It was intentionally done that way to make it painfully obvious to remove it when the filter hook is changed.

You are right though. I was split on the idea and you thought of it too, so I'll change it.

**Comment by arnaudbroes** [comment]
This is good now.

**Pull request merged**